src/libcore/macros/mod.rs

```diff
@@ -711,6 +711,47 @@ macro_rules! todo {
     ($($arg:tt)+) => (panic!("not yet implemented: {}", $crate::format_args!($($arg)+)));
 }
 
+/// Extracts the successful type of a `Poll<T>`.
+///
+/// This macro bakes in propagation of `Pending` signals by returning early.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(ready_macro)]
+///
+/// use core::task::{Context, Poll};
+/// use core::future::Future;
+/// use core::pin::Pin;
+///
+/// async fn get_num() -> usize {
+///     42
+/// }
+///
+/// pub fn do_poll(cx: &mut Context<'_>) -> Poll<()> {
+///     let mut f = get_num();
+///     let f = unsafe { Pin::new_unchecked(&mut f) };
```

Comment:
I don't think the safety of this is obvious to users who are not familiar with `Pin`. Could you replace this with safe way (use `Box::pin`, or rewrite this example to use `future::ready()`)?

Reply:
Ah yes for sure — when I wrote this `future::ready` did not exist in std yet, but now that it does it seems like an obvious match. Will do!